app/src/vesselInfo/vesselInfoActions.js

```diff
@@ -96,39 +98,20 @@ function setCurrentVessel(tilesetId, seriesgroup, fromSearch) {
   return (dispatch, getState) => {
     const state = getState();
     const token = state.user.token;
-    let request;
-
-    if (typeof XMLHttpRequest !== 'undefined') {
-      request = new XMLHttpRequest();
-    } else {
-      throw new Error('XMLHttpRequest is disabled');
-    }
-
     const layer = state.layers.workspaceLayers.find(l =>
       LAYER_TYPES_WITH_HEADER.indexOf(l.type) > -1 && l.tilesetId === tilesetId);
 
-    request.open(
-      'GET',
-      buildEndpoint(layer.header.endpoints.info, { id: seriesgroup }),
-      true
-    );
-    if (token) {
-      request.setRequestHeader('Authorization', `Bearer ${token}`);
-    }
-    request.setRequestHeader('Accept', 'application/json');
-    request.onreadystatechange = () => {
-      if (request.readyState !== 4) {
-        return;
-      }
-      if (request.status >= 500 || request.status === 404) {
-        console.error('Error loading vessel info:', request.responseText);
-        return;
-      }
-      const data = JSON.parse(request.responseText);
+    const vesselInfoUrl = buildEndpoint(layer.header.endpoints.info, { id: seriesgroup });
+    fetchEndpoint(vesselInfoUrl, token).then((data) => {
       delete data.series;
-
       data.tilesetId = tilesetId;
 
+      // mocking API comments to show warning
+      // TODO:
+      //  1. Check field name from API
+      //  2. Remove the next line of code
+      data.comment = 'This vessel has a warning';
```

Comment:
@j8seangel I would remove this now so we can merge the PR, once the API starts delivering the comment it should work.

Reply:
Sounds, good, it is there to test the feature so I will remove it before merging.